pkg/cortex/modules.go

```diff
@@ -638,7 +642,7 @@ var modules = map[moduleName]module{
 	},
 
 	TableManager: {
-		deps: []moduleName{Server},
+		deps: []moduleName{Server, Ring, Ingester},
```

Comment:
This does not look right. It looks like `TableManager` would now also start an `Ingester`.

Maybe we could create `KVStore` client and do CAS operations to claim a key with heartbeat mechanism. One who holds the key would be allowed to do table operations. Consider key to be `tablemanager` and value to include an ID(could be hostname or some other random key) and `update-time`, which would be treated as heartbeat expected to be updated by tablemanager owning the key, every time it ticks [here](https://github.com/cortexproject/cortex/blob/master/pkg/chunk/table_manager.go#L219).

I was thinking of doing it like:
For each tick [here](https://github.com/cortexproject/cortex/blob/master/pkg/chunk/table_manager.go#L219) do CAS as follow: 
a. check if ID is mine, if yes update the heartbeat
b. else check if `update-time` is too old, if yes update the key with own id and the current time.
If CAS was successful do the table operations.

@pracucci @jtlisi what do you guys think about this approach?